src/optimizer/query_planner.c

```diff
@@ -9331,27 +9535,54 @@ static double
 qo_range_selectivity (QO_ENV * env, PT_NODE * pt_expr)
 {
   PT_NODE *lhs, *arg1, *arg2;
-  PRED_CLASS pc1;
+  PRED_CLASS pc1, pc2;
   double total_selectivity, selectivity;
-  int lhs_icard, rhs_icard, icard;
+  int lhs_icard = 0, rhs_icard = 0, icard = 0;
```

Comment:
if these variables are set 1, you don't need to check whether it is calculating at first. e.g.` if (lhs_icard == 0)` at the following. In my opinion, this change makes code more clear.

Reply:
I don't think so. '0' means the variable is not set. Changing it to '1' can make the source code hard to read.